src/pybind/mgr/dashboard/frontend/src/app/ceph/shared/feedback/feedback.component.ts

```diff
@@ -0,0 +1,119 @@
+import { FormControl, Validators } from '@angular/forms';
+
+import { NgbActiveModal, NgbModal } from '@ng-bootstrap/ng-bootstrap';
+
+import { ActionLabelsI18n } from '~/app/shared/constants/app.constants';
+import { CdFormGroup } from '~/app/shared/forms/cd-form-group';
+
+@Component({
+  selector: 'cd-feedback',
+  templateUrl: './feedback.component.html',
+  styleUrls: ['./feedback.component.scss']
+})
+export class FeedbackComponent {
+  title = 'Feedback';
+  project: any = [
+    'dashboard',
+    'block',
+    'objects',
+    'file_system',
+    'ceph_manager',
+    'orchestrator',
+    'ceph_volume',
+    'core_ceph'
+  ];
+  tracker: any = ['bug', 'feature'];
+  api_key: any;
+
+  feedbackForm: CdFormGroup;
+  tracker_key_req: boolean;
+  constructor(
+    public activeModal: NgbActiveModal,
+    public actionLabels: ActionLabelsI18n,
+    public secondaryModal: NgbModal
+  ) {
+    this.createForm();
+    this.check_key();
+  }
+
+  private createForm() {
+    this.feedbackForm = new CdFormGroup({
+      project: new FormControl('', Validators.required),
+      tracker: new FormControl('', Validators.required),
+      subject: new FormControl('', Validators.required),
+      description: new FormControl('', Validators.required),
+      api_key: new FormControl('')
+    });
+  }
+
+  private check_key() {
+    fetch('/api/feedback/api_key', {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        'Accept': 'application/vnd.ceph.api.v1.0+json',
+        'Accept-Language': 'en-US,en;q=0.5',
+        'Connection': 'keep-alive'
+      }
+    })
+      .then((res) => res.json())
+      .then((res) => {
+        this.tracker_key_req = res['issue_key_exists'] ? false : true;
+      });
+  }
+
+  async onSubmit() {
+    if (this.tracker_key_req) {
+      await fetch('/api/feedback/api_key', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'Accept': 'application/vnd.ceph.api.v1.0+json',
+          'Accept-Language': 'en-US,en;q=0.5',
+          'Connection': 'keep-alive'
+        },
+        body: JSON.stringify({ 'api_key': this.feedbackForm.controls['api_key'].value })
+      });
+    }
+    await fetch('/api/feedback', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'Accept': 'application/vnd.ceph.api.v1.0+json',
+        'Accept-Language': 'en-US,en;q=0.5',
+        'Connection': 'keep-alive'
+      },
+      body: JSON.stringify({
+        'project': this.feedbackForm.controls['project'].value,
+        'tracker': this.feedbackForm.controls['tracker'].value,
+        'subject': this.feedbackForm.controls['subject'].value,
+        'description': this.feedbackForm.controls['description'].value
+      })
+    })
+      .then((res) => {
+        if (res.status === 401) {
+          return Promise.reject('Invalid API key');
+        } else {
+          return res.json();
+        }
+      })
```

Comment:
Regarding the API interaction, I still suggest you to move that code to a separate service. You can take as an example the following service for managing users: https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/frontend/src/app/shared/api/user.service.ts